src/Slices/DesiredState/UI/Page.tsx

```diff
@@ -34,76 +41,106 @@ export const Page: React.FC = () => {
-      >
-        <TableControls
-          filter={filter}
-          setFilter={setFilter}
-          paginationWidget={
-            <PaginationWidget
-              data={data}
-              pageSize={pageSize}
-              setPageSize={setPageSize}
-              setCurrentPage={setCurrentPage}
-            />
-          }
-        />
-        <ToastAlert
-          data-testid="ToastAlert"
-          title={words("desiredState.actions.promote.failed")}
-          message={errorMessage}
-          setMessage={setErrorMessage}
-        />
-        <RemoteDataView
-          data={data}
-          retry={retry}
-          label="DesiredStatesView"
-          SuccessView={(desiredStates) =>
-            desiredStates.data.length <= 0 ? (
-              <EmptyView
-                message={words("desiredState.empty.message")}
-                aria-label="DesiredStatesView-Empty"
-              />
-            ) : (
-              <DesiredStatesTable
-                rows={desiredStates.data}
-                aria-label="DesiredStatesView-Success"
+  useEffect(() => {
+    if (deleteVersion.isError) {
+      setErrorMessage(deleteVersion.error.message);
+    }
+  }, [deleteVersion.isError, deleteVersion.error]);
+
+  if (isError) {
+    return (
+      <ErrorView
+        data-testid="ErrorView"
+        title={words("error")}
+        message={words("error.general")(error.message)}
+        ariaLabel="DesiredStatesView-Failed"
+        retry={refetch}
+      />
+    );
+  }
+
+  if (isSuccess) {
+    const handlers =
+      typeof data.links === "undefined"
+        ? {}
+        : getPaginationHandlers(data.links, data.metadata);
+
+    return (
+      <PageContainer pageTitle={words("desiredState.title")}>
+        <GetDesiredStatesContext.Provider
+          value={{
+            filter,
+            pageSize,
+            currentPage,
+            setErrorMessage,
+            compareSelection,
+            setCompareSelection,
+            setDeleteModal,
+          }}
+        >
+          <TableControls
+            filter={filter}
+            setFilter={setFilter}
+            paginationWidget={
+              <PaginationWidget
+                data={RemoteData.success({
+                  handlers,
+                  metadata: data.metadata,
+                })}
+                pageSize={pageSize}
+                setPageSize={setPageSize}
+                setCurrentPage={setCurrentPage}
               />
-            )
-          }
-        />
-        <DeleteModal
-          kind="DeleteVersion"
-          version={versionToDelete}
-          isOpened={isModalOpened}
-        />
-      </GetDesiredStatesContext.Provider>
-    </PageContainer>
-  );
+            }
+          />
+          <ToastAlert
+            data-testid="ToastAlert"
+            title={words("desiredState.actions.promote.failed")}
+            message={errorMessage}
+            setMessage={setErrorMessage}
+          />
+          {data.data.length <= 0 ? (
```

Comment:
How come you end up with data.data?

Reply:
That's how result from the endpoint looks like
in the useGetDesiredStates there is interface but basically the endpoints returns:
{
  data: DesiredStateVersion[];
  links: Pagination.Links;
  metadata: Pagination.Metadata;
}